### PR TITLE
feat(includers/sourcedocs): includer implementation

### PR DESCRIPTION
**Pull request by moki**
sourcedocs includer implementation

### CODE REVIEW

File: src/services/tocs.ts [hunk 1]
```diff
@@ -68,6 +70,50 @@ function add(path: string) {
         parsedToc.title = _liquidSubstitutions(parsedToc.title, combinedVars, path);
     }
 
+    for (const item of parsedToc.items) {
```
Comment (yndx-birman) [inline src/services/tocs.ts:73]: Let's move this logic to a separate function,`_replaceIncludesByIncluders` maybe?

File: src/services/tocs.ts [hunk 2]
```diff
@@ -68,6 +70,50 @@ function add(path: string) {
...
+            content: content && typeof content === 'object' ? dump(content) : content,
+            path: path.replace(rootInput, inputFolderPath),
+        }) as {content: string; path: string});
+
+        // eslint-disable-next-line no-shadow
+        for (const {content, path} of prepared) {
+            writeFileSync(path, content);
+        }
+
+        item.include.path = join(item.include.path, 'toc.yaml');
```
Comment (yndx-birman) [inline src/services/tocs.ts:114]: Let's check if `include.path` already ends on `toc.yaml`

File: src/services/includers/batteries/sourcedocs.ts [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+const isMdVisible = allPass([isMdExtension, complement(isHidden)]);
+
+function generateContent(params: IncluderFnParams): IncluderFnOutput {
+    const {
+        include: {path, mode},
+        root,
+    } = params;
+
+    if (mode !== 'link') {
+        throw new Error('include with the includer supports only link mode, set mode: link');
```
Comment (yndx-birman) [inline src/services/includers/batteries/sourcedocs.ts:18]: Where will it be caught?

File: src/services/tocs.ts [hunk 4]
```diff
@@ -68,6 +70,50 @@ function add(path: string) {
         parsedToc.title = _liquidSubstitutions(parsedToc.title, combinedVars, path);
     }
 
+    for (const item of parsedToc.items) {
+        if (!item?.include?.includer) {
+            continue;
+        }
+
+        if (!isValidIncluder(item.include)) {
+            logger.info(resolve(inputFolderPath, path), `includer: ${item.include.includer} not implemented`);
```
Comment (yndx-birman) [inline src/services/tocs.ts:79]: In quite mode we won't see this message if includer is not implemented. Let's leave only logger.error

Comment (moki) [reply src/services/tocs.ts:79]: gonna add message passing to error logger

File: src/services/tocs.ts (hunk 2 above)
Comment (moki) [reply src/services/tocs.ts:114]: gonna add path check: allowed path is directory
do it together with mode: link,  where link is allowed and no mode treated as link, and wrong mode is fatal error.

File: src/services/tocs.ts (hunk 1 above)
Comment (moki) [reply src/services/tocs.ts:73]: gonna move it to the apply includers funcdtion

File: src/services/includers/batteries/sourcedocs.ts (hunk 3 above)
Comment (moki) [reply src/services/includers/batteries/sourcedocs.ts:18]: gonna add generic error handling in the apply includers

**Review by yndx-birman — APPROVED** [review]

**Pull request merged**